proposal.rst

```diff
@@ -239,7 +301,82 @@ Project Timeline
+	- Start working on GSOC task:  "Rework Omega templates to use more modern web techniques"
+
+Week 3 
+	- Continue working on Omega templates.
+	- Identify areas that requires change user-end APIs, consult with mentors about this.
+	- Continue working on the user-end APIs.
+
+Week 4
+	- Continue working on the user-end APIs.
+	- Add tests to see consistency of new API(). 
+
+MONTH 1 DELVERABLE : 
+	[MINUMUM] : New "Practical Example" and "Installation" on Xapian Docs. Omega supporting templates (Subtask completed)	
+	[SATISFACTORY] : Above + New-API prototype + API-testing. Such that API is fully covered with unittests and use-cases.
+	
+
+MONTH 2  : 
+
+BIG TASK NAME : DOCUMENTATION STEP + REPORTING-SYSTEM
+DELEVARABLE OR SUBMITTABLE : 
+
+Week 1 
+	- Buffer week if things are not completed from previous month otherwise continue. 
+
+Week 2 
+	- Identify the requirements of a good reporting system(metrics, benchmarks, which algorithms etc..) as consulted by the developers on IRC and mentors
+	- Read and go through xapian-letor/tests and ranker/scorer where actual examples are given of Marching-Learning Pipeline(data-grabbing,training,storing Qrel..,scoring)
+
+Week 3 
+	- Develop a basic prototype of the charts and graph using a Jupyter notebook. This is good to see what types of visualisations suits the numbers produced
+	- Study the comparison and benchmarking strategies between different ranking algs include in the front-end(xapian-evaluate a good place to start)
+	- Test the implementation with INEX and see the number matching to "IR Evaluation of Letor Ranking Scheme" in https://trac.xapian.org/wiki/GSoC2011/LTR/Notes#IREvaluationofLetorrankingscheme. 
+
+Week 4
+	- Document the new Algorithm Benchmarking system on Xapian Docs
+	- Explore the feasibility of "Xapian backend for advanced trac search" with TracAdvancedSearchPlugin
+	- If due-diligence succeeds, work for next two weeks on it. 
+
+MONTH 2 DELEVERABLE : 
+	[MINIMUM] : basic front-end ready for visualizing benchmark numbers + tested the implementation of benchmark system on cli.
+	[SATISFIABLE] : Above + some progress in TracAdvancedSearchPlugin
+
+MONTH 3  : 
+
+BIG TASK NAME : REPORTING-SYSTEM + FINISHING/CONCLUSION 
+
+Week 1 
+	- Buffer week if things are not completed from previous month otherwise continue. 
+	- Xapian backend support in TracAdvancedSearchPlugin
+
+Week 2 
+	- Xapian backend support in TracAdvancedSearchPlugin
+	- Start investigation make dist for letor, to ensure and get first release version.
+
+Week 3 
+	-  Buffer week if things are not completed from previous month otherwise continue.
+
+Week 4
+	-  Buffer week if things are not completed from previous month otherwise continue.
+
+MONTH 2 DELEVERABLE : 
+
+	[MINIMUM] : Started with Xapian backend support with TracAdvancedSearchPlugin and documented the procedure for future follow-ups.
+	[SATISFIABLE] : Fully integreate Xapian backend with TracAdvancedSearchPlugin and complete all the preceding projects.	
```

Comment:
I assume this is actually meant to be "MONTH 3 DELIVERABLE"?

I'm concerned that you're trying to cram far too many different things into a 3 month project, without having actually yet dug into many of them to any great extent to get an idea for how much work they might actually involved - for example, I don't think we've seen any patches from you yet.

There's a common tendency for GSoC students to try to over-promise (perhaps in part thinking that might make their proposal look more appealing to the org, but I suspect mostly due to youthful optimism) but it's much better to start with a realistic project scope rather than have to start hacking off large chunks of the timeline when it becomes clear that it's not all going to happen.  This is why we tell you that you need to start to dig into the code that you'd working on, and to break down the tasks into subtasks until you get to tasks that you can actually usefully estimate the time for.  If one just thinks at a high level, one inevitably overlooks complicated details that will take a lot of the time.

I think you need to decide which part you want to work on most and focus on that.